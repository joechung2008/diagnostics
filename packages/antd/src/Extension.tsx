import { Typography } from "antd";
import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div className="extension-root grow">
      <Typography.Title level={1}>{extensionName}</Typography.Title>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
