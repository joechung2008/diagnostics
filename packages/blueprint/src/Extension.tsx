import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div id="extension" className="extension-root grow">
      <h2 className="bp6-heading">{extensionName}</h2>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
