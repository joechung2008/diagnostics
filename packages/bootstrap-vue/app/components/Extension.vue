<template>
  <div class="extension">
    <h1>{{ extensionName }}</h1>
    <Configuration v-if="config" :config="config" />
    <StageDefinition v-if="stageDefinition" :stage-definition="stageDefinition" />
  </div>
</template>

<script setup lang="ts">
import type { ExtensionProps } from "@/types/Extension";

defineProps<ExtensionProps>();
</script>

<style scoped>
.extension {
  display: flex;
  flex: 1;
  flex-direction: column;
  gap: 0.5rem;
  max-height: max-content;
  overflow-y: auto;
  padding: 0.25rem;
}
</style>
