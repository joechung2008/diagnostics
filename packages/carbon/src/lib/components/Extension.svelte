<script lang="ts">
	import Configuration from './Configuration.svelte';
	import StageDefinition from './StageDefinition.svelte';

	let { extensionName, config, stageDefinition }: ExtensionProps = $props();
</script>

<div class="extension-root" role="tabpanel" tabindex="0">
	<h1>{extensionName}</h1>
	{#if config}
		<Configuration {config} />
	{/if}
	{#if stageDefinition}
		<StageDefinition {stageDefinition} />
	{/if}
</div>

<style>
	.extension-root {
		display: flex;
		flex: 1;
		flex-direction: column;
		gap: 0.5rem;
		overflow-y: auto;
		padding: 0.5rem;
	}
</style>
