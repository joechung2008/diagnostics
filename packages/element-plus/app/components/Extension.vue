<template>
    <div class="extension">
        <h1>{{ extensionName }}</h1>
        <Configuration v-if="config" :config="config" />
        <StageDefinition v-if="stageDefinition" :stage-definition="stageDefinition" />
    </div>
</template>

<script setup lang="ts">
import type { ExtensionProps } from "@/types"

defineProps<ExtensionProps>()
</script>

<style scoped>
.extension {
    display: flex;
    flex: 1;
    flex-direction: column;
    gap: 0.5rem;
    max-height: max-content;
    overflow-y: auto;
    padding: 0.25rem;
}

h1 {
    margin: 0;
}
</style>
