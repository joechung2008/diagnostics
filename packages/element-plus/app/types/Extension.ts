import type { ExtensionInfo } from "./ExtensionInfo"
import type { ExtensionError } from "./ExtensionError"

export type Extension = ExtensionInfo | ExtensionError
