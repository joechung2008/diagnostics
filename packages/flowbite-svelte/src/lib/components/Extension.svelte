<script lang="ts">
  import { Heading } from "flowbite-svelte";
  import Configuration from "./Configuration.svelte";
  import StageDefinition from "./StageDefinition.svelte";

  let { config, extensionName, stageDefinition }: ExtensionProps = $props();
</script>

<div class="flex flex-grow flex-col gap-0.5 overflow-y-auto p-1">
  <Heading tag="h1">
    {extensionName}
  </Heading>
  {#if config}
    <Configuration {config} />
  {/if}
  {#if stageDefinition}
    <StageDefinition {stageDefinition} />
  {/if}
</div>
