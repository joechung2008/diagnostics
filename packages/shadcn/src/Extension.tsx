import Configuration from "@/Configuration";
import StageDefinition from "@/StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div className="flex flex-col gap-2 max-h-max overflow-y-auto p-2 flex-grow">
      <h1 className="text-xl font-semibold">{extensionName}</h1>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
