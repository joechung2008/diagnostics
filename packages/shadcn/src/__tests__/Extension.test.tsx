import Extension from "@/Extension";
import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

describe("Extension", () => {
  it("matches snapshot with full data (config + stageDefinition)", () => {
    const props = {
      extensionName: "sample-extension",
      config: {
        FOO: "bar",
        BAZ: "qux",
      },
      stageDefinition: {
        build: ["compile", "unit-test"],
        release: ["staging", "production"],
      },
    };
    const { asFragment } = render(<Extension {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot with only extensionName", () => {
    const props = {
      extensionName: "minimal-extension",
    };
    const { asFragment } = render(<Extension {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
