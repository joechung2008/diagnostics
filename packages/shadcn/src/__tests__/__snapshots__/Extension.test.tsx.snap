// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Extension > matches snapshot with full data (config + stageDefinition) 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-2 max-h-max overflow-y-auto p-2 flex-grow"
  >
    <h1
      class="text-xl font-semibold"
    >
      sample-extension
    </h1>
    <div>
      <h2
        class="text-lg font-semibold"
      >
        Configuration
      </h2>
      <div
        class="relative w-full overflow-x-auto"
        data-slot="table-container"
      >
        <table
          aria-label="Configuration"
          class="w-full caption-bottom text-sm"
          data-slot="table"
        >
          <thead
            class="[&_tr]:border-b"
            data-slot="table-header"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                Key
              </th>
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                Value
              </th>
            </tr>
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0"
            data-slot="table-body"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                FOO
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                bar
              </td>
            </tr>
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                BAZ
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                qux
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div>
      <h2
        class="text-lg font-semibold"
      >
        Stage Definitions
      </h2>
      <div
        class="relative w-full overflow-x-auto"
        data-slot="table-container"
      >
        <table
          aria-label="Stage Definitions"
          class="w-full caption-bottom text-sm"
          data-slot="table"
        >
          <thead
            class="[&_tr]:border-b"
            data-slot="table-header"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                Key
              </th>
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                Value
              </th>
            </tr>
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0"
            data-slot="table-body"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                build
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                compile, unit-test
              </td>
            </tr>
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                release
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                staging, production
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Extension > matches snapshot with only extensionName 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-2 max-h-max overflow-y-auto p-2 flex-grow"
  >
    <h1
      class="text-xl font-semibold"
    >
      minimal-extension
    </h1>
  </div>
</DocumentFragment>
`;
